ui/src/components/PostSpinner.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from "react";
+import styled from "styled-components";
+import MoonLoader from "react-spinners/MoonLoader";
+
+const LoaderBlock = styled.div`
+  width: 100%;
+  height: 50vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const LoaderText = styled.span`
+  display: block;
+  margin-top: 1rem;
+  color: var(--color-darkblue);
+  font-weight: 600;
+`;
+
+const PostSpinner: React.FC = function () {
+  return (
+    <LoaderBlock>
+      <div>
```

Comment:
Don't need this extra `<div>`. Apply `flex-direction: column` to LoaderBlock instead.